### ISSUE TITLE
fix: @Expose() class decorator not working

### ISSUE THREAD

**Issue by alexvallecit** [typestack/class-transformer#1170]
### Description
When I apply the `@Expose()` decorator to each property of my class and do the `plainToInstance` it works correctly, it transforms the properties of the class as I expect it to.
But if I apply the `@Expose()` decorator to the class, it does not work as I expect, it does not transform the properties of the class.

**Minimal code-snippet showcasing the problem**
```ts
@Expose()
class Data {
  foo: string;

  bar: number;
}

console.log('data: ',  plainToInstance(Data, { foo: 'foo', bar: '1' }))
```

### Expected behavior
`data: {
  "foo": "foo",
  "bar": 1
}`

### Actual behavior
`data: {
  "foo": "foo",
  "bar": "1"
}`

**Comment by ChoGathK** [comment]
This is because `@Expose` is a property decorator, not a class decorator.
@alexvallecit

**Comment by spiv** [comment]
If this is true, the comment here is deceptive:
https://github.com/typestack/class-transformer/blob/develop/src/decorators/expose.decorator.ts#L5

**Comment by Branchverse** [comment]
> If this is true, the comment here is deceptive: https://github.com/typestack/class-transformer/blob/develop/src/decorators/expose.decorator.ts#L5

I totally agree, have the same issue, I can imagine they maybe meant class-properties? but then usally if you place a decorator over an entire class they say it cannot be applied here which is not the case.

And I don't like redundant code, it would be highly appreciated if this could be fixed

**Comment by Branchverse** [comment]
Actually I managed to make it work as intended, but it needs somehting that hasn't been said before:
The decorator`@Type(() => ClassType)`  seem to be the key here.
Once I specified the Type of my subclasses they got entirely exposed with the `@Expose` above the entire class
